src/repositories/indexable-cleanup-repository.php

```diff
@@ -239,6 +239,54 @@ public function clean_indexables_for_non_publicly_viewable_taxonomies( $limit )
 		// phpcs:enable
 	}
 
+	/**
+	 * Cleans up any indexables that belong to post type archive page that are not/no longer publicly viewable.
+	 *
+	 * @param int $limit The limit we'll apply to the queries.
+	 *
+	 * @return bool|int The number of deleted rows, false if the query fails.
+	 */
+	public function clean_indexables_for_non_publicly_viewable_post_type_archive_pages( $limit ) {
+		global $wpdb;
+		$indexable_table = Model::get_table_name( 'Indexable' );
+
+		$included_post_types = $this->post_type->get_indexable_post_archives();
+
+		$post_archives = [];
+
+		foreach ( $included_post_types as $post_type ) {
+			$post_archives[] = $post_type->name;
+		}
+		// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared -- Reason: Too hard to fix.
+		if ( empty( $included_post_types ) ) {
```

Comment:
I think you mean `if ( empty( $post_archives) ) {`.

I'm gonna push it, please give a thumbs up if you agree